docs/source/conf.py

```diff
@@ -440,9 +448,16 @@ def setup(app):
         import binascii
         paramiko.client.hexlify = lambda x: binascii.hexlify(x).decode()
         paramiko.util.safe_string = lambda x: '' # function result never *actually used*
-    class EndlessLoop(Exception): pass
-    def alrm_handler(sig, frame):
-        signal.alarm(180) # three minutes
+
+class EndlessLoop(Exception): pass
+
+if 'doctest' in sys.argv:
+    q = Queue()
+    t = Thread(target=launch_test, args=())
+    t.daemon = True
+    t.start()
+
+    try:
+        t.join(timeout=180)
+    except:
         raise EndlessLoop()
-    signal.signal(signal.SIGALRM, alrm_handler)
-    signal.alarm(600) # ten minutes
```

Comment:
The whole purpose of `EndlessLoop` was to raise it inside the hanging context in order to know what the hanging doctest examples are. Please revert.

Reply:
I see that you meant a different timeout mechanism under windows. You can just add an `if not windows` for these two `signal.signal(...); signal.alarm(...)` lines

Reply:
I have set up a signle mechanism working for both windows and linux. Are you sure it is not better than handle any case for any os? If really you are sure I can do what you want.